tests/virt_autotest/virsh_internal_snapshot.pm

```diff
@@ -35,6 +35,9 @@ sub run_test {
     foreach my $guest (keys %xen::guests) {
         my $type = check_guest_disk_type($guest);
         next if ($type == 1);
+        record_info "virsh-snapshot", "Cleaning in case of rerun";
+        assert_script_run "virsh snapshot-delete $guest --snapshotname internal-snapshot-$guest-01 || true";
+        assert_script_run "virsh snapshot-delete $guest --snapshotname internal-snapshot-$guest-02 || true";
```

Comment:
With this, "virsh snapshot-delete" errors, for the case of snapshot 01/02 exist but deleting meets failure, will be ignored. And when this happens, although it can continue to following tests, it will fail them anyway. And the tester will need to look back to find the root cause.

What about changing it to logic like "if virsh snapshot-list $guest|grep internal-snapshot-$guest-01;then virsh snapshot-delete $guest --snapshotname internal-snapshot-$guest-01" with assert_script_run?

Reply:
The deleting part of the test is on the line 58 - this is at the beginning to ensure that the environment is clean for the test.

Reply:
Same here: Why not use `script_run` instead?

Reply:
Did so. Thank you

Reply:
> The deleting part of the test is on the line 58 - this is at the beginning to ensure that the environment is clean for the test.

Similarly as last reply. Yes, you are not testing it, so I understand that you do not plan to care about its result from test points perspective. But it is a necessary preparation step if the snapshot actually exists before starting the cared test steps, isn't it? If yes, assert_script_run   is the api to select.

Reply:
```perl
if (script_run("virsh snapshot-list $guest | grep internal-snapshot-$guest-01") != 0) {
    assert_script_run "virsh snapshot-delete $guest --snapshotname internal-snapshot-$guest-01 || true";
}
if (script_run("virsh snapshot-list $guest | grep internal-snapshot-$guest-02") != 0) {
    assert_script_run "virsh snapshot-delete $guest --snapshotname internal-snapshot-$guest-02 || true";
}
```
@alice-suse what do you say about it now?